src/components/pages/blocks/latest-blocks-table.tsx

```diff
@@ -0,0 +1,112 @@
+import Link from 'next/link';
+import { l2PublicClient } from "@/lib/chains";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { formatTimestamp } from "@/lib/utils";
+import { Progress } from '@/components/ui/progress';
+
+interface LatestBlocksTableProps {
+  latestBlockNumber: bigint;
+  currentPage: number;
+  blocksPerPage: number;
+}
+
+async function fetchBlocks(latestBlockNumber: bigint, currentPage: number, blocksPerPage: number) {
+  const startBlock = latestBlockNumber - BigInt((currentPage - 1) * blocksPerPage);
+  const blocks = await Promise.all(
+    Array.from({ length: blocksPerPage }, (_, i) =>
+      l2PublicClient.getBlock({ blockNumber: startBlock - BigInt(i) })
```

Comment:
There's an edge case here when we're hitting the last page: if there's less than 25 blocks left to display, this code tries to get blocks with an invalid negative block number.
Make sure to guard against this edge case and display only the < 25 remaining blocks.

Reply:
Alright. Will do that ASAP. Thanks for your kind review.